_topics/en/op_checktemplateverify.md

```diff
@@ -139,15 +139,16 @@ assures several receivers that they can each be paid.  This two-step
 process can probably be used anywhere payment batching is an option
 but it can likely reduce fees even further than payment batching.
 
-Later versions of the proposal placed greater emphasis on other
+The original proposal also emphasized other
 contracts and [covenants][topic covenants] that could be created using
 the new opcode, such as the ability to create [channel
-factories][topic channel factories], [vaults][topic vaults], and
+factories][topic channel factories], [vaults][topic vaults],
 [coinjoin transactions][topic coinjoin] in new ways that might
-simplify construction or reduce fees.  Other authors have mentioned
-that the new opcode could possibly be used to allow users to
+simplify construction or reduce fees, and described how
+the new opcode could possibly be used to allow users to
 trustlessly [pool their funds][joinpool] together into a single UTXO
-in a way that would increase privacy.
+in a way that would increase privacy. These other concepts have been
+described in further detail in subsequent works by a variety of authors.
```

Comment:
> Gregory Maxwell [discussed](http://gnusha.org/bitcoin-wizards/2019-05-21.log) some of his first impressions of the COSHV proposal to the logged IRC channel #bitcoin-wizards. Among them was a description of how the above multi-transaction tree could be used as part of a coinjoin construction to manage balances offline.

this text still seems accurate to me, so i don't see the descrepency? He gave a description of how it could be used, it doesn't same "greg came up with a whole new way to use it not mentioned in the BIP at all".

Reply:
I've only quickly re-skimmed the original COSHV proposal, but I  don't see it mentioning what we now call joinpools/payment pools/coinpools.  It does describe channel factories, which are pretty much joinpools used for second-layer setup transactions, and it also describes the advantages to single-tx coinjoins, which is obviously related to joinpools---but I don't see anything that describes the novelty of using shared UTXO ownership with keypath spends for vastly enhanced onchain privacy and COSHV for non-interactive creation of the unilateral exit option.

Reply:
I think the below text is to that effect:

> Each node of the tree can also attempt to 'opt in' to preferring a Taproot signature based spend, but if participants are offline or malicious the expansion can proceed to smaller groups.

> If desired, the Tapscript path can be usurped by a signature based spend to improve fungibility.


> Critically, because this is a Tapscript, it is intended that participants may collaborate to replace the Tapscript path with a signature. This lifts the requirement of the output being spent alone and the exact match of output hashes, if other dependencies (like channel state) can be updated.